pyoozie/model.py

```diff
@@ -159,38 +170,37 @@ def suspended(cls):
             return [status for status in cls if status.is_suspended()]
 
         def is_unknown(self):
-            return self.value.status_id == 0
+            return self._value_.status_id == 0
 
         def is_active(self):
-            return self.value.is_active
+            return self._value_.is_active
 
         def is_running(self):
-            return self.value.is_running
+            return self._value_.is_running
 
         def is_suspendable(self):
-            return self.value.is_suspendable
+            return self._value_.is_suspendable
 
         def is_suspended(self):
-            return self.value.is_suspended
+            return self._value_.is_suspended
 
     def __init__(self, oozie_api, details, parent=None):
         self._oozie_api = oozie_api
         self._parent = parent
-        self.toString = None
         details = dict(details)
-        for key, func in self.REQUIRED_KEYS.iteritems():
+        for key, func in self.REQUIRED_KEYS.items():
             value = details.pop(key, None)
             try:
-                parsed_value = func(self, value) if func and value is not None else value
+                parsed_value = func(self, value) if func else value
```

Comment:
Subtle change: always call the parse function, even for `None`. This allows me the opportunity to tweak the default value; e.g. `status` should default to `UNKNOWN` if unspecified.